static/app/actionCreators/organization.tsx

```diff
@@ -132,49 +132,55 @@ export function fetchOrganizationDetails(
     PageFiltersStore.onReset();
   }
 
-  const loadOrganization = async () => {
-    try {
-      await fetchOrg(api, slug, usePreload);
-    } catch (err) {
-      if (!err) {
-        return;
-      }
+  const loadOrganization = () => {
+    return new Promise(async resolve => {
+      let org: Organization | undefined = undefined;
+      try {
+        org = await fetchOrg(api, slug, usePreload);
+      } catch (err) {
+        if (!err) {
+          return;
+        }
 
-      OrganizationStore.onFetchOrgError(err);
+        OrganizationStore.onFetchOrgError(err);
 
-      if (err.status === 403 || err.status === 401) {
-        const errMessage =
-          typeof err.responseJSON?.detail === 'string'
-            ? err.responseJSON?.detail
-            : typeof err.responseJSON?.detail?.message === 'string'
-              ? err.responseJSON?.detail.message
-              : null;
+        if (err.status === 403 || err.status === 401) {
+          const errMessage =
+            typeof err.responseJSON?.detail === 'string'
+              ? err.responseJSON?.detail
+              : typeof err.responseJSON?.detail?.message === 'string'
+                ? err.responseJSON?.detail.message
+                : null;
```

Comment:
nit: can we destruct / simplify this ternary? Is there a helper to read API error messages already?

Reply:
i pulled it out into a function